### PR TITLE
BGP-triggered GUE Test

### PR DESCRIPTION
**Pull request by wenovus**

### CODE REVIEW

File: integration_tests/twodut_oneotg_tests/bgp_triggered_gue/bgp_triggered_gue_test.go [hunk 1]
```diff
@@ -0,0 +1,624 @@
...
+			t.Fatalf("ERROR: Could not create temporary pcap file: %v\n", err)
+		}
+		defer os.Remove(f.Name())
+
+		if _, err := f.Write(captureBytes); err != nil {
+			t.Fatalf("ERROR: Could not write bytes to pcap file: %v\n", err)
+		}
+		f.Close()
+
+		handleRead, err := pcap.OpenOffline(f.Name())
```
Comment (DanG100) [inline integration_tests/twodut_oneotg_tests/bgp_triggered_gue/bgp_triggered_gue_test.go:501]: use https://pkg.go.dev/github.com/google/gopacket/pcapgo instead, it doesn't rely on libpcap being install on the system

File: integration_tests/twodut_oneotg_tests/bgp_triggered_gue/bgp_triggered_gue_test.go [hunk 2]
```diff
@@ -0,0 +1,624 @@
...
+		PeerAs:          ygot.Uint32(dutAS),
+		NeighborAddress: ygot.String(dutPort3.IPv4),
+	})
+	gnmi.Replace(t, dut, bgpPath.Config(), dutConf)
+	gnmi.Replace(t, dut2, bgpPath.Config(), dut2Conf)
+
+	nbrPath := bgpPath.Neighbor(dut2Port2.IPv4)
+	gnmi.Await(t, dut, nbrPath.SessionState().State(), 60*time.Second, oc.Bgp_Neighbor_SessionState_ESTABLISHED)
+
+	testTrafficAndPrintCapture := func(t *testing.T, title string, startingIP string, encapFields *EncapFields) {
```
Comment (DanG100) [inline integration_tests/twodut_oneotg_tests/bgp_triggered_gue/bgp_triggered_gue_test.go:476]: can this be a moved outside this func? would help readability

File: integration_tests/twodut_oneotg_tests/bgp_triggered_gue/bgp_triggered_gue_test.go [hunk 3]
```diff
@@ -0,0 +1,624 @@
...
+				}
+				if diff := cmp.Diff(wantTL, gotTL, cmpopts.IgnoreUnexported(layers.UDP{}), cmpopts.IgnoreFields(layers.UDP{}, "BaseLayer")); diff != "" {
+					t.Errorf("network layer (-want, +got):\n%s", diff)
+				}
+				// TODO: Check that lower layers is the original packet.
+			}
+		}
+	}
+
+	testTrafficAndPrintCapture(t, "without policy", "198.51.0.0", nil)
```
Comment (DanG100) [inline integration_tests/twodut_oneotg_tests/bgp_triggered_gue/bgp_triggered_gue_test.go:575]: feels like this whole could made in table-style test

**Comment by DanG100** [comment]
LGTM after lint fix

**Review by DanG100 — APPROVED** [review]

**Comment by wenovus** [comment]
I'm going to merge this because everything passes locally -- GitHub actions is giving a strange error -- almost as if it only picked up half of the "fix lint" commit.

**Pull request merged**